src/selection.ts

```diff
@@ -14,6 +17,33 @@ export const getRange = (selection: Selection): Range => {
   return range;
 };
 
+export const cleanSelection = (selection: Selection): Selection => {
+  const anchor = selection.anchorNode;
+  const focus = selection.focusNode;
+
+  if (!anchor || !focus) {
+    return selection;
+  }
+
+  const range = getRange(selection);
+
+  // if selection starts w/ iframe or range startContainer is an img, replace anchorNode with its parent
+  // fixes firefox behavior that prevented starting highlights on iframes or images without previous siblings
```

Comment:
this is making more sense now, but i forget why the firefox behavior was actually preventing starting highlights in this case. was this the same as the snapping text issue fixed by your changes below? or was there something else?

Reply:
iframe selection has different behavior from img selection in Firefox. A selection that begins and/or ends on an iframe uses the <iframe> element as its anchor and/or focus rather than looking for a nearby text node sibling. The only solution I found for this was setting the iframe parent as anchor/focus (which is what's used on Chrome and I think Safari).